src/server/game/Chat/ChatCommands/ChatCommandArgs.h

```diff
@@ -99,6 +102,84 @@ struct ArgInfo<std::string, void>
     }
 };
 
+// enum
+template <typename T>
+struct ArgInfo<T, std::enable_if_t<std::is_enum_v<T>>>
+{
+    typedef typename std::underlying_type<T>::type uType;
+
+    static inline std::map<std::string, Optional<T>> MakeSearchMap()
+    {
+        std::map<std::string, Optional<T>> map;
+        for (T val : EnumUtils::Iterate<T>())
+        {
+            EnumText text = EnumUtils::ToString(val);
+
+            std::string title = std::string(text.Title);
+            strToLower(title);
```

Comment:
i wonder if this should be treated as a wstr... doesn't matter right now, but what if someone puts silly characters in a title one of these days?